nodejs/matrix/node.js

```diff
@@ -36,44 +36,51 @@ module.exports = function(item, subnet) {
     }
   }
 
-  for (var i = 1; i < flow.length; i++) { // i=0 is header;
-    step = flow[i][0];
+  for (var flow_at_timestep_idx = 1; flow_at_timestep_idx < flow.length; flow_at_timestep_idx++) { // i=0 is header;
+    step = flow[flow_at_timestep_idx][0];
     time = new Date(step).getTime();
+
     // Get boundary Conditions
 
+    // if we're inside the time-based subset that we're supposed to process
     if ((!config.start || config.start < time) &&
       (!config.stop || time < config.stop)) {
 
-      steps.push(flow[i][0]);
+      // attach the current timestep to the output
+      steps.push(flow[flow_at_timestep_idx][0]);
 
       // Add Inflows from edge links
       // debugger;
       for (e = 0; e < inbound.length; e++) {
         edge = inbound[e].properties;
-        if( edge.flow.length >= i ) continue;
+        //if( edge.flow.length >= flow_at_timestep_idx ) continue; // commented out as part of issue #1 in UCM repo
         
         rows.push([
           u.id('INBOUND', step),
           u.id(p.prmname, step),
           e, 
           0, 
           1, 
-          u.roundBound(edge.flow[i][1]), 
-          u.roundBound(edge.flow[i][1])
+          u.roundBound(edge.flow[flow_at_timestep_idx][1]),
+          u.roundBound(edge.flow[flow_at_timestep_idx][1])
         ]);
       }
+
+      // Attach outbound boundary conditions
       for (e = 0; e < outbound.length; e++) {
         edge = outbound[e].properties;
-        if( edge.flow.length >= i ) continue;
```

Comment:
@msdogan and @josue-medellin just calling your attention to this. I'd made this fix to the calvin-network-tools code back in 2020 to make regional exports/subnets work, but it involves moving/removing some code that I don't full understand the implications of. Can you take a look at the changes here and see if you think they're OK and that regional/subnet exports are still safe and the results are OK, or if we should undo them?